### ISSUE TITLE
Doesn't work with Android 8

### ISSUE THREAD

**Issue by danny005** [arleyandrada/PushClient#178]
I have compiled this module with SDK 7.x and it works fine with Android 7 and lower.

Unfortunately there is a problem with Android 8. The app crashs if it is not running while receiving a push. And with the app in background, nothing happens (no visible push notification).

Has anyone managed to get this module working with Android 8?

**Comment by danny005** [comment]
Ok, figured it out. NotificationChannel is missing.

**Comment by ErikDohmen** [comment]
@danny005 Did you manage to get this working? Cause it keeps crashing on my side even though I've added NotificationChannels

**Comment by danny005** [comment]
Nope. I had to compile with android:targetSdkVersion="25" as a workaround.

**Comment by JuananSpain** [comment]
hello,
I have activated the developer option "Show notifications channel notification", and I have found that in api 26 when a notification is received, this notification does not appear, and it shows me the following channel warning "failed to post notification on channel null" , other notification appcelerator modules have the same problem. When I put targetSdkVersion = "25" this does not happen.

**Comment by ErikDohmen** [comment]
Anyone any progress? In november the switch needs to be made, but there's no way still to fix this for as far as I know.

**Comment by JuananSpain** [comment]
I finally gave up, I chose to use the libraries on Android:
Ti.playservices (11.0.40)
firebase.core
firebase.cloudmessaging
I have been able to compile with version 26. The parameters of the push when the app is in background were obtained as extras in the "onresume".

**Comment by ErikDohmen** [comment]
@JuananSpain you only switched to the firebase.cloudmessaging for android and still use it for ios?
Any guidelines to keep in mind when switching?

**Comment by JuananSpain** [comment]
Yes, for now I manage android and Ios separately, later I imagine that with time I will end up implementing it for ios too.
